### PR TITLE
Facebook login here we go

### PR DESCRIPTION
**Pull request by eklemen**

### CODE REVIEW

**Comment by eklemen** [comment]
i can login with facebook and it creates a new object in firebase with the users ID, trying to grab the user name from within the controller.
Auth factory in index.js
controller is group/group.controller.js

File: src/app/index.js [hunk 1]
```diff
@@ -21,11 +21,68 @@ angular.module('studyBuddy', ['restangular', 'ui.router', 'firebase'])
...
+            fb: authdUser.facebook,
+            uid: authdUser.facebook.id,
+            name: authdUser.facebook.displayName
+        });
+        
+        user = $firebaseObject(
+            auth.child(authdUser.facebook.id)
+        );
+        return user;
+    }
```
Comment (al-the-x) [inline src/app/index.js:95]: I am now sorry I ever gave this code to anyone.

File: src/app/group/group.controller.js [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
+    });
+//    var myGroup = self.user.child('groupInfo');
+//    console.log(myGroup);
+    var ref = new Firebase('https://study-buddy.firebaseio.com/users/' + self.user.$id);
+    var ref2 = $firebaseObject(new Firebase('https://study-buddy.firebaseio.com/users/' + self.user.$id))
+    ref2.$loaded()
+    .then(function(data){
+        self.name = data.name
+    })
+    this.obj = $firebaseArray(ref);
```
Comment (al-the-x) [inline src/app/group/group.controller.js]: I detect much confusion here... when `onAuth` completes `self.user` will be the user object for the authenticated user. Attempting to load that user as a `$firebaseArray` doesn't really make sense. I can see what you're trying to do with it afterward... maybe you want a `groups` property of that user object loaded as a `$firebaseArray` instead?

Comment (eklemen) [inline src/app/group/group.controller.js]: Oh man, this was committed mid changes, oops! Yeah the last commit, i have the group info being submitted under the user. I was having trouble grabbing info from the facebook object. It looked as though i could just follow the trail into the object putting .facebook.displayName but was getting undefined on everything except the self.user. I will commit what the code should have looked like

**Pull request merged**